src/RegisterPage.cpp

```diff
@@ -135,17 +155,24 @@ RegisterPage::RegisterPage(QWidget *parent)
         top_layout_->addLayout(form_wrapper_);
         top_layout_->addStretch(1);
         top_layout_->addLayout(button_layout_);
-        top_layout_->addStretch(1);
         top_layout_->addWidget(error_label_, 0, Qt::AlignHCenter);
+        top_layout_->addStretch(1);
 
         connect(back_button_, SIGNAL(clicked()), this, SLOT(onBackButtonClicked()));
         connect(register_button_, SIGNAL(clicked()), this, SLOT(onRegisterButtonClicked()));
 
         connect(username_input_, SIGNAL(returnPressed()), register_button_, SLOT(click()));
+        connect(username_input_, &TextField::editingFinished, this, [this]() { checkFields(); });
         connect(password_input_, SIGNAL(returnPressed()), register_button_, SLOT(click()));
+        connect(password_input_, &TextField::editingFinished, this, [this]() { checkFields(); });
         connect(password_confirmation_, SIGNAL(returnPressed()), register_button_, SLOT(click()));
+        connect(
+          password_confirmation_, &TextField::editingFinished, this, [this]() { checkFields(); });
         connect(server_input_, SIGNAL(returnPressed()), register_button_, SLOT(click()));
-        connect(this, &RegisterPage::registerErrorCb, this, &RegisterPage::registerError);
+        connect(server_input_, &TextField::editingFinished, this, [this]() { checkFields(); });
+        connect(this, &RegisterPage::registerErrorCb, this, [this](const QString &msg) {
+                showError(msg);
+        });
```

Comment:
```suggestion
        connect(this, &RegisterPage::registerErrorCb, this, &RegisterPage::showError);
```

Reply:
Ah, no idea, if that actually works. It may not compile, since showError is overloaded, didn't see that.

Reply:
There are troubles with overloaded, and when I tried to fix it with qoverload<>, it's was not compile

Reply:
Sure, then use the lambda :3